frontend/metrics.go

```diff
@@ -98,16 +98,16 @@ func (mm MetricsMiddleware) Metrics() MiddlewareFunc {
 				"route":       routePattern,
 				"state":       string(sub.State),
 			})
+		} else {
+			mm.Emitter.EmitCounter("frontend_count", 1.0, map[string]string{
+				"verb":        r.Method,
+				"api_version": r.URL.Query().Get(APIVersionKey),
+				"code":        strconv.Itoa(lrw.statusCode),
+				"route":       routePattern,
+				"state":       "Unknown",
+			})
 		}
 
-		// Emit metrics
-		mm.Emitter.EmitCounter("frontend_count", 1.0, map[string]string{
-			"verb":        r.Method,
-			"api_version": r.URL.Query().Get(APIVersionKey),
-			"code":        strconv.Itoa(lrw.statusCode),
-			"route":       routePattern,
-		})
-
```

Comment:
I think we can simplify that a bit if we remove the duplication of code about the Emitter part and just assign the appropriate value to the subscriptionState, what about something like this:
```go
                subscriptionState := "Unknown"
		if subscriptionId != "" {
			sub, exists := mm.cache.GetSubscription(subscriptionId)

			if !exists {
				arm.WriteError(
					w, http.StatusBadRequest,
					arm.CloudErrorInvalidSubscriptionState, "",
					UnregisteredSubscriptionStateMessage,
					subscriptionId)
				return
			}
			subscriptionState = string(sub.State)
		}
		
		mm.Emitter.EmitCounter("frontend_count", 1.0, map[string]string{
			"verb":        r.Method,
			"api_version": r.URL.Query().Get(APIVersionKey),
			"code":        strconv.Itoa(lrw.statusCode),
			"route":       routePattern,
			"state":       subscriptionState,
		})
```

Reply:
Sure, adjusted!